src/lavinmq/amqp/client.cr

```diff
@@ -413,16 +413,23 @@ module LavinMQ
       private def close_socket
         @running = false
         @socket.close
-        @log.debug { "Socket closed" }
       rescue ex
         @log.debug { "#{ex.inspect} when closing socket" }
       end
 
-      def close(reason = nil)
+      def close(reason = nil, timeout : Time::Span = 5.seconds)
         reason ||= "Connection closed"
-        @log.info { "Closing, #{reason}" }
+        @log.info { "Closing: #{reason}" }
+
         send AMQP::Frame::Connection::Close.new(320_u16, "CONNECTION_FORCED - #{reason}", 0_u16, 0_u16)
         @running = false
+
+        socket = @socket
+        if socket.responds_to?(:"write_timeout=")
+          socket.write_timeout = timeout
+        end
```

Comment:
shouldn't the write timeout be before the actual send? 

How about also setting a `read_timeout`? So that the read_loop will close the connection even if it doesn't receive a frame (because the client is overloaded or something).

Reply:
> How about also setting a `read_timeout`? So that the read_loop will close the connection even if it doesn't receive a frame (because the client is overloaded or something).
Yeah, probably. I didn't because I think we always set a read timeout.